src/data.c

```diff
@@ -1482,6 +1480,9 @@ int pr_data_xfer(char *cl_buf, size_t cl_size) {
           pr_pool_tag(tmp_pool, "ASCII download");
         }
 
+        /* Fill up our internal buffer. */
+        memcpy(session.xfer.buf, cl_buf, buflen);
```

Comment:
We only want/need to do this `memcpy(2)` _iff_ we're going to be manipulating the contents of the buffer, as during ASCII translation.